### ISSUE TITLE
Support for yml short file extension

### ISSUE THREAD

**Issue by caphrim007** [Sceptre/sceptre#852]
I noticed while walking through the tutorial that sceptre makes the assumption that yaml files will end with the `.yaml` file extension. Looking at the code seems to confirm this.

So I was interested in soliciting the maintainers to see what their tolerance would be for adding the short-form version of YAML extensions (`.yml`) to the supported list.

I wouldn't mind taking a shot at implementing it, but wouldn't want to start if it wouldn't be accepted.

**Comment by 1oglop1** [comment]
Hi @caphrim007 Thank you for opening the issue.
I have raised this discussion few times myself internally, this question is probably old as YAML itself.
The main problem is that spec failed to specify and recommend the extension from the beginning.
As you can see on this "hidden" site https://yaml.org/faq.html they recommended yaml (2001-2006)

You can also find numerous Stack Overflow questions on this topic discussing the popularity of `.yml` vs `.yaml`.

The reason we decided to stick with enforcing `.yaml` is to simply keep everything consistent by providing less options. 
This way you avoid having mix of `.yml` and `.yaml` files and avoid having to check for duplicated configs, etc.

We are open to evaluate this decision in case the `.yaml` extension causes any trouble on a system you are using Sceptre on.

**Comment by craighurley** [comment]
IMO: if the spec says it's `.yaml`, it seems reasonable to stick to `.yaml`.

**Comment by caphrim007** [comment]
@1oglop1 I get it. I was coming from the Ansible world where they can't seem to make up their mind on which extension to use, so they kinda support both. There are some areas where they require `.yml` (for instance their Ansible Galaxy tool requires it) and since I was mostly playing in their sandbox, I just obliged and switched all my files over to use that.

In my environment, I wrap Sceptre in other tools though so I can control the file extension. I will use `.yaml` for my Sceptre work.

This is why I asked before I went off and started making a PR. Thanks for the clarification and history behind the decision. I prefer convention over configuration anyway, so I can't really argue with the decision.
